### DISCUSSION TITLE
Any updates on real time data stream (Websockets)?

### DISCUSSION THREAD

**Discussion by charlie-niekirk** [Noxcrew/mcchampionship-api#6] [General]
It's mentioned that this is something you may consider but hadn't decided on yet, is there any update on this? I'd like to build a mobile app (as an open source career portfolio project) and real time event info using Websockets would be awesome! Seems like the site (mcc.live) listens to firebase Firestore/Realtime Database change events to update the page so not sure how difficult it'd be to build a Websocket server to expose this event stream.

**Comment by MagicA550** [comment]
Yep, you're correct; it is something we're considering internally for the future. During the inter-season break we're taking a once-over on our web services (both on the user-facing side and the back-end) with a couple of changes already scheduled in for the API, but I can't confirm these as of yet as they're still pending confirmation from elsewhere.

As for the possibility of WebSockets, exposing a live data stream is something I'm looking into for the future but I cannot say for certain whether it will happen in the near future or, indeed, ever. I do agree that they are certainly more efficient than a polling system and it shouldn't be too difficult to provide data streams for the endpoints we're already serving.

Thanks for the question, feel free to get in touch for anything else! :)

**Comment by charlie-niekirk** [comment]
Hi there! Have there been any more thoughts on this recently? Thanks :)
